integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleBigQueryTest.java

```diff
@@ -0,0 +1,551 @@
+    final int testId = 1;
+
+    String sourceTableName = SOURCE_TABLE_NAME_TEMPLATE + testId;
+    String destinationTableName = SINK_TABLE_NAME_TEMPLATE + testId;
+
+    Table sourceTable = createTestTable(bigQueryDataset, sourceTableName, getSimpleFieldsSchema());
+    insertData(bigQueryDataset, sourceTableName, Paths.get(SOURCE_PATH + "simple_source.json"));
+
+    Assert.assertFalse(isExisting(destinationTableName));
+
+    io.cdap.cdap.api.data.schema.Schema sourceSchema = getSimpleTableSchem();
+
+    Map<String, String> sourceProps = new ImmutableMap.Builder<String, String>()
+      .put("referenceName", "bigQuery_source")
+      .put("project", getProjectId())
+      .put("dataset", bigQueryDataset)
+      .put("table", sourceTableName)
+      .put("schema", sourceSchema.toString())
+      .put("serviceFilePath", getServiceAccountCredentials())
+      .build();
+
+    // insert data into new Kind within the same Namespace using Auto-generated key and custom indexes
+    Map<String, String> sinkProps = new ImmutableMap.Builder<String, String>()
+      .put("referenceName", "bigQuery_sink")
+      .put("project", getProjectId())
+      .put("dataset", bigQueryDataset)
+      .put("table", destinationTableName)
+      .put("allowSchemaRelaxation", "false")
+      .put("serviceFilePath", getServiceAccountCredentials())
+      .build();
+
+    int expectedCount = 1;
+
+    List<FieldValue> expectedResult = getResultTableDate(TableId.of(bigQueryDataset, sourceTableName));
+    com.google.cloud.bigquery.Schema expectedSchema = sourceTable.getDefinition().getSchema();
+
+    GoogleBigQueryTest.DeploymentDetails deploymentDetails =
+      deployApplication(sourceProps, sinkProps, BIG_QUERY_PLUGIN_NAME + "-storeInNewTable");
+    startWorkFlow(deploymentDetails.getAppManager(), ProgramRunStatus.COMPLETED);
+
+    ApplicationId appId = deploymentDetails.getAppId();
+    Map<String, String> tags = ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, appId.getNamespace(),
+                                               Constants.Metrics.Tag.APP, appId.getEntityName());
+
+    checkMetric(tags, "user." + deploymentDetails.getSource().getName() + ".records.out", expectedCount, 10);
+    checkMetric(tags, "user." + deploymentDetails.getSink().getName() + ".records.in", expectedCount, 10);
+
+    Assert.assertTrue(isExisting(destinationTableName));
+
+    com.google.cloud.bigquery.Schema resultSchema = getTableSchema(TableId.of(bigQueryDataset, destinationTableName));
+    Assert.assertEquals(expectedSchema, resultSchema);
+
+    List<FieldValue> result = getResultTableDate(TableId.of(bigQueryDataset, destinationTableName));
+    Assert.assertEquals(expectedResult.size(), result.size());
+
+    for (int i = 0; i < result.size(); i++) {
+      Assert.assertEquals(expectedResult.get(i).getAttribute(), result.get(i).getAttribute());
+      Assert.assertEquals(expectedResult.get(i).getValue(), result.get(i).getValue());
```

Comment:
Can we throw a bit more informative exception here. Like which attribute has different values.

Also I see you check these in loop is that guaranteed that the FieldValues are in the same order?